### PR TITLE
Updating documentation for new datetime format

### PR DESCRIPTION
**Pull request by CaptainDredge**
### Description
This PR updates documentation for new datetime format `rfc_3339_lenient` 

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/6308

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
Thanks, @CaptainDredge ! Did you mean to link this to issue https://github.com/opensearch-project/documentation-website/issues/6308? Is the content ready for doc review?

**Comment by CaptainDredge** [comment]
> Thanks, @CaptainDredge ! Did you mean to link this to issue https://github.com/opensearch-project/documentation-website/issues/6308? Is the content ready for doc review?

Thanks @hdhalter I've linked to the right issue. The content is ready for review but I'm not able to figure out what exact formatting is required for CI to go green. Is there any automated linter for the docs?

File: _field-types/supported-field-types/date.md [hunk 1]
```diff
@@ -137,6 +137,7 @@ Format name and description | Pattern and examples
 `year_month_day` <br> `strict_year_month_day` <br> A four-digit year, two-digit month, and two-digit day. | "yyyy-MM-dd" <br> "2019-03-23" 
 `year_month` <br> `strict_year_month` <br> A four-digit year and two-digit month. | "yyyy-MM" <br> "2019-03" 
 `year` <br> `strict_year` <br> A four-digit year. | "yyyy" <br> "2019" 
+`rfc3339_lenient` <br>An RFC3339 compatible datetime formatter which is much faster than other full date lenient formats like `strict_date_optional_time` | Multiple patterns. <br> "YYYY" <br> "2019" <br> "YYYY-MM" <br> "2019-03" <br> "YYYY-MM-DD" <br> "2019-03-23" <br> "YYYY-MM-DDThh:mmTZD" <br> "2019-03-23T21:34Z" <br> "YYYY-MM-DDThh:mm:ssTZD" <br> "2019-03-23T21:34:46Z" <br> "YYYY-MM-DDThh:mm:ss.sTZD" <br> "2019-03-23T21:34:46.123456789-04:00" <br> "YYYY-MM-DDThh:mm:ss,sTZD" "2019-03-23T21:34:46,123456789-04:00"
```
Comment (Naarcha-AWS) [inline _field-types/supported-field-types/date.md:140]:
```suggestion
`rfc3339_lenient` <br>An RFC3339 compatible DateTimeFormatter which is much faster than other full date-lenient formats like `strict_date_optional_time` | "YYYY" <br> "2019" <br> "YYYY-MM" <br> "2019-03" <br> "YYYY-MM-DD" <br> "2019-03-23" <br> "YYYY-MM-DDThh:mmTZD" <br> "2019-03-23T21:34Z" <br> "YYYY-MM-DDThh:mm:ssTZD" <br> "2019-03-23T21:34:46Z" <br> "YYYY-MM-DDThh:mm:ss.sTZD" <br> "2019-03-23T21:34:46.123456789-04:00" <br> "YYYY-MM-DDThh:mm:ss,sTZD" "2019-03-23T21:34:46,123456789-04:00"
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

**Review by Naarcha-AWS — APPROVED** [review]

**Comment by Naarcha-AWS** [comment]
Moving to done, we should update Vale rules so that an `emdash` isn't always required.

**Comment by kolchfa-aws** [comment]
Endash is required in text but this is literal strings so they should be in code format. Let me make some suggestions.

File: _field-types/supported-field-types/date.md [hunk 2]
```diff
@@ -137,6 +137,7 @@ Format name and description | Pattern and examples
 `year_month_day` <br> `strict_year_month_day` <br> A four-digit year, two-digit month, and two-digit day. | "yyyy-MM-dd" <br> "2019-03-23" 
 `year_month` <br> `strict_year_month` <br> A four-digit year and two-digit month. | "yyyy-MM" <br> "2019-03" 
 `year` <br> `strict_year` <br> A four-digit year. | "yyyy" <br> "2019" 
+`rfc3339_lenient` <br>An RFC3339 compatible DateTimeFormatter which is much faster than other full date-lenient formats like `strict_date_optional_time` | "YYYY" <br> "2019" <br> "YYYY-MM" <br> "2019-03" <br> "YYYY-MM-DD" <br> "2019-03-23" <br> "YYYY-MM-DDThh:mmTZD" <br> "2019-03-23T21:34Z" <br> "YYYY-MM-DDThh:mm:ssTZD" <br> "2019-03-23T21:34:46Z" <br> "YYYY-MM-DDThh:mm:ss.sTZD" <br> "2019-03-23T21:34:46.123456789-04:00" <br> "YYYY-MM-DDThh:mm:ss,sTZD" "2019-03-23T21:34:46,123456789-04:00"
```
Comment (kolchfa-aws) [inline _field-types/supported-field-types/date.md:140]:
```suggestion
`rfc3339_lenient` <br>An RFC3339 compatible DateTimeFormatter which is much faster than other full date-lenient formats like `strict_date_optional_time` | `"YYYY"` <br> `"2019"` <br> `"YYYY-MM"` <br> `"2019-03"` <br> `"YYYY-MM-DD"` <br> `"2019-03-23"` <br> `"YYYY-MM-DDThh:mmTZD"` <br> `"2019-03-23T21:34Z"` <br> `"YYYY-MM-DDThh:mm:ssTZD"` <br> `"2019-03-23T21:34:46Z"` <br> `"YYYY-MM-DDThh:mm:ss.sTZD"` <br> `"2019-03-23T21:34:46.123456789-04:00"` <br> `"YYYY-MM-DDThh:mm:ss,sTZD"` <br> `"2019-03-23T21:34:46,123456789-04:00"`
```

**Review by kolchfa-aws — COMMENTED** [review]
Looking at this, we should change all string literals to have code format. @CaptainDredge If you'd like to take this on, that would be great. Otherwise, we can address this in a separate issue. Thanks!

File: _field-types/supported-field-types/date.md [hunk 3]
```diff
@@ -99,70 +99,71 @@ Components of basic date formats are not separated by a delimiter. For example,
...
-`date_hour_minute` <br> `strict_date_hour_minute` <br> A full date, two-digit hour, and two-digit minute. | "yyyy-MM-dd`T`HH:mm" <br> "2019-03-23T21:34"
-`date_hour` <br> `strict_date_hour` <br> A full date and two-digit hour, separated by `T`. | "yyyy-MM-dd`T`HH" <br> "2019-03-23T21" 
-`date` <br> `strict_date` <br> A four-digit year, two-digit month, and two-digit day. | "yyyy-MM-dd" <br> "2019-03-23" 
-`year_month_day` <br> `strict_year_month_day` <br> A four-digit year, two-digit month, and two-digit day. | "yyyy-MM-dd" <br> "2019-03-23" 
-`year_month` <br> `strict_year_month` <br> A four-digit year and two-digit month. | "yyyy-MM" <br> "2019-03" 
-`year` <br> `strict_year` <br> A four-digit year. | "yyyy" <br> "2019" 
+`date_optional_time`<br>`strict_date_optional_time` <br> A generic full date and time. Year is required. Month, day, and time are optional. Time is separated from date by `T`. | Multiple patterns. <br>"2019-03-23T21:34:46.123456789-04:00" <br> `"2019-03-23T21:34:46"` <br> `"2019-03-23T21:34"` <br> `"2019"`
```
Comment (Naarcha-AWS) [inline _field-types/supported-field-types/date.md:127]:
```suggestion
`date_optional_time`<br>`strict_date_optional_time` <br> A generic full date and time. Year is required. Month, day, and time are optional. Time is separated from date by `T`. | Multiple patterns. <br>`"2019--03--23T21:34:46.123456789--04:00"` <br> `"2019-03-23T21:34:46"` <br> `"2019-03-23T21:34"` <br> `"2019"`
```

**Pull request merged**